firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -177,20 +180,27 @@ public InstallationResponse createFirebaseInstallation(
   private void writeFIDCreateRequestBodyToOutputStream(
       HttpURLConnection httpURLConnection, @NonNull String fid, @NonNull String appId)
       throws IOException {
-    OutputStream outputStream = httpURLConnection.getOutputStream();
-    if (outputStream == null) {
-      throw new IOException(
-          "Cannot send CreateInstallation request to FIS. No OutputStream available.");
-    }
-
-    GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream);
     try {
-      gzipOutputStream.write(
-          buildCreateFirebaseInstallationRequestBody(fid, appId).toString().getBytes("UTF-8"));
+      writeRequestBodyToOutputStream(
+          httpURLConnection, getJsonBytes(buildCreateFirebaseInstallationRequestBody(fid, appId)));
     } catch (JSONException e) {
       throw new IllegalStateException(e);
-    } finally {
-      gzipOutputStream.close();
+    }
+  }
+
+  private static byte[] getJsonBytes(JSONObject jsonObject) throws UnsupportedEncodingException {
```

Comment:
I wonder if this creates more confusion than clarity. What do you think?

Maybe it's better to just say `throws IOException` and explain in the JavaDoc (or even an inline comment) that it is a `UnsupportedEncodingException`?

Reply:
Done.